.dotnet/tests/TestScenarios/Assistants/AssistantTests.cs

```diff
@@ -585,10 +583,92 @@ public void BasicFileSearchWorks()
                 }
             }
         }
-        Assert.That(messages.Count > 1);
+        Assert.That(messages.Count() > 1);
         Assert.That(messages.Any(message => message.Content.Any(content => content.Text.ToLower().Contains("cake"))));
     }
 
+    [Test]
+    public async Task CanEnumerateAssistants()
+    {
+        AssistantClient client = GetTestClient();
+
+        // Create assistant collection
+        for (int i = 0; i < 10; i++)
+        {
+            Assistant assistant = client.CreateAssistant("gpt-3.5-turbo");
+            Validate(assistant);
+            Assert.That(assistant.Name, Is.Null.Or.Empty);
+        }
+
+        List<Assistant> allAssistants = new();
+
+        // Page through collection
+        int count = 0;
+        AsyncPageableCollection<Assistant> assistants = client.GetAssistantsAsync();
+
+        await foreach (Assistant assistant in assistants)
+        {
+            Console.WriteLine($"[{count,3}] {assistant.Id} {assistant.CreatedAt:s} {assistant.Name}");
+
+            count++;
+
+            allAssistants.Add(assistant);
+        }
+
+        // Clean up
+        foreach (Assistant assistant in allAssistants)
+        {
+            client.DeleteAssistant(assistant.Id);
+        }
+
+        Assert.AreEqual(10, count);
+    }
+
+    [Test]
+    public async Task CanPageThroughAssistantCollection()
+    {
+        AssistantClient client = GetTestClient();
+
+        // Create assistant collection
+        for (int i = 0; i < 10; i++)
+        {
+            Assistant assistant = client.CreateAssistant("gpt-3.5-turbo");
+            Validate(assistant);
+            Assert.That(assistant.Name, Is.Null.Or.Empty);
+        }
+
+        List<Assistant> allAssistants = [];
+
+        // Page through collection
+        int count = 0;
+        int pageCount = 0;
+        AsyncPageableCollection<Assistant> assistants = client.GetAssistantsAsync();
+        IAsyncEnumerable<ResultPage<Assistant>> pages = assistants.AsPages(pageSizeHint: 2);
+
+        await foreach (ResultPage<Assistant> page in pages)
+        {
+            foreach (Assistant assistant in page)
+            {
+                Console.WriteLine($"[{count,3}] {assistant.Id} {assistant.CreatedAt:s} {assistant.Name}");
+
+                allAssistants.Add(assistant);
+
+                count++;
+            }
+
+            pageCount++;
+        }
+
+        // Clean up
+        foreach (Assistant assistant in allAssistants)
+        {
+            client.DeleteAssistant(assistant.Id);
```

Comment:
Won't this end up deleting every Assistant in the organization? If it's just a matter of cleanup, the `Validate(assistant)` line accompanying creation should take care of it -- it adds the ID to a test cleanup step to delete the temporary assistants.

In a shared org, `foreach (Assistant in client.GetAssistants()) { client.DeleteAssistant(assistant); }` could be very bad :)

Reply:
Let me piggyback on existing cleanup methods instead of my clearly troublesome bespoke thing!

Reply:
Addressed, but LMK if it still looks wrong!